Installation instructions (#28238)
ArduPilot/ardupilot: README.md

## Installation Instructions

> Follow these steps to clone the ArduPilot repository, set up your environment, and configure it for the CubeOrangeXL-DID autopilot.

Reviewer: ```suggestion
Follow these steps to clone the ArduPilot repository, set up your environment, and configure it for a specific flight controller hardware board.
```

> ### Prerequisites

Reviewer: ```suggestion
### Prerequisites for Linux
```

Before proceeding, ensure that the following tools and libraries are installed on your system:

1. **Git** (for cloning the repository):
   ```bash
   sudo apt update
   sudo apt install git
   ```

2. **Python 3** (used by `waf` build system):
   ```bash
   sudo apt install python3 python3-pip
   ```

> 3. **ARM GCC Toolchain** (required to compile ArduPilot for ARM-based boards like CubeOrange):

Reviewer: ```suggestion
3. **ARM GCC Toolchain** (required to compile ArduPilot for ARM-based boards):
```

   ```bash

>    sudo apt install gcc-arm-none-eabi gdb-arm-none-eabi

Reviewer 2: that isn't our toolchain. don't use it

   ```

4. **Symlink Python 3 to Python**:
   Some scripts expect `python` to point to Python 3. Create a symlink to ensure compatibility:
   ```bash
   sudo ln -s /usr/bin/python3 /usr/bin/python
   ```

### Clone the Repository

1. **Clone the repository with the required branch**:

>    Clone the `pt3.1-dcp` branch, including all submodules:
>    ```bash
>    git clone --branch pt3.1-dcp --recurse-submodules https://github.com/Reblade/ardupilot.git

Reviewer: ```suggestion
   Usually you either want a stable branch like `Copter-4.5` or `Plane-4.5` or you want the latest development branch `master`.
   Clone the desired branch, including all submodules, `Copter-4.5` in this example:
   ```bash
   git clone --branch Copter-4.5 --recurse-submodules https://github.com/ArduPilot/ardupilot.git
```

   ```

2. **Change to the cloned directory**:
   Navigate into the `ardupilot` directory:
   ```bash
   cd ardupilot
   ```

### Configure the Build

> 1. **Configure the build system for the CubeOrangeXL-DID board**:
>    ```bash
>    ./waf configure --board CubeOrangeXL-DID

Reviewer: ```suggestion
1. **Configure the build system for the desired flight controller board, in this example the `CubeOrangeXL-DID` board**:
   ```bash
   ./waf configure --board CubeOrangeXL-DID
```

   ```

   This step will set up the environment and ensure that the build system is properly configured for the target autopilot board.

### Build and Upload Firmware (Optional)

> Once configured, you can build and upload the firmware to the CubeOrangeXL autopilot:

Reviewer: ```suggestion
Once configured, you can build and upload the firmware to the flight controller:
```

1. **Build the copter firmware**:
   ```bash
   ./waf copter
   ```

2. **Upload the firmware to the autopilot** (if connected via USB):
   ```bash
   ./waf copter --upload
   ```

> Now you have successfully cloned, configured, and built the ArduPilot firmware for the CubeOrangeXL-DID autopilot.

Reviewer: ```suggestion
Now you have successfully cloned, configured, and built the ArduPilot firmware for your flight controller.
```
